src/VirtoCommerce.Platform.Hangfire/Middleware/HangfireUserContextMiddleware.cs

```diff
@@ -0,0 +1,54 @@
+using System.Threading;
+using Hangfire.Client;
+using Hangfire.Server;
+using Microsoft.AspNetCore.Http;
+using static VirtoCommerce.Platform.Core.Common.ThreadSlotNames;
+
+namespace VirtoCommerce.Platform.Hangfire.Middleware
+{
+    /// <summary>
+    /// This class allow to process all hangifre jobs to add user name from identity and save it to
+    /// the Thread after job is perfoming to achieve getting access to user name in background tasks
+    /// </summary>
+    public class HangfireUserContextMiddleware : IClientFilter, IServerFilter
+    {
+        private readonly IHttpContextAccessor _contextAccessor;
+
+        public HangfireUserContextMiddleware(IHttpContextAccessor contextAccessor) => _contextAccessor = contextAccessor;
+
+        private string ContextUserName => _contextAccessor?.HttpContext?.User?.Identity?.Name;
+
+        #region IClientFilter Members
+
+        public void OnCreating(CreatingContext filterContext)
+        {
+            if (ContextUserName is null)
+                return;
+
+            filterContext.SetJobParameter(USER_NAME, ContextUserName);
+        }
+
+        public void OnCreated(CreatedContext filterContext)
+        {
+            // Pass
+        }
+
+        #endregion IClientFilter Members
+
+        #region IServerFilter Members
+
+        public void OnPerforming(PerformingContext filterContext)
+        {
+            var userName = filterContext.GetJobParameter<string>(USER_NAME);
+
+            Thread.SetData(Thread.GetNamedDataSlot(USER_NAME), userName);
```

Comment:
When testing the Orders change event (e.g. change Approved and save) I've not got the proper user in [PlatformOperationLog] table.
After debugging found out that it relates to the fact that thread when this middleware.OnPerforming is executed differs from the one where DbTriggers are executed https://github.com/VirtoCommerce/vc-platform/blob/5d51ae50fddded58be04a454db8daa65e2b94518/src/VirtoCommerce.Platform.Data/Extensions/ApplicationBuilderExtensions.cs#L16, so passing data using this mechanism is not working here.